app/app.go

```diff
@@ -425,10 +452,35 @@ func NewEvmos(
 		app.AccountKeeper, app.BankKeeper, scopedTransferKeeper,
 	)
 	transferModule := transfer.NewAppModule(app.TransferKeeper)
+	transferIBCModule := transfer.NewIBCModule(app.TransferKeeper)
+
+	app.ICAControllerKeeper = icacontrollerkeeper.NewKeeper(
+		appCodec, keys[icacontrollertypes.StoreKey], app.GetSubspace(icacontrollertypes.SubModuleName),
+		app.IBCKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
+		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+		scopedICAControllerKeeper, app.MsgServiceRouter(),
+	)
+
+	app.ICAHostKeeper = icahostkeeper.NewKeeper(
+		appCodec, keys[icahosttypes.StoreKey], app.GetSubspace(icahosttypes.SubModuleName),
+		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+		app.AccountKeeper, scopedICAHostKeeper, app.MsgServiceRouter(),
+	)
+
+	icaModule := ica.NewAppModule(&app.ICAControllerKeeper, &app.ICAHostKeeper)
+
+	// initialize ICA module with mock module as the authentication module on the controller side
```

Comment:
So, I think you don't need to set up the ICA controller module if you aren't planning on having a custom ICA Auth module. The mock module is just used for writing tests within `ibc-go`.

Or did you have some other use case? 

cc: @colinaxner

Reply:
Setting up the ICA host module will allow other chains to create interchain accounts on evmos. You will need to whitelist all the message types that an interchain account can execute. This can be done via on chain params (gov) or updating genesis file. 

If you want to set up a custom ICA auth module (this allows evmos to be a controller chain) see this basic example in the ICA repo -> https://github.com/cosmos/interchain-accounts/tree/master/x/inter-tx . We have some more updated demos coming soon. We're currently waiting on the gov module update so we can demo gov/ica functionality from a controller chain perspective (gov proposal to control ICA account registered by evmos on other chains). 

Once that is done you'll want to update the router also see -> https://github.com/cosmos/interchain-accounts/blob/master/app/app.go#L380

Reply:
@seantking do you see any issues regarding using different keys or HD paths for an account (we use the key derivation from ethereum secp256k1 keys)

Reply:
No, using different keys shouldn't be an issue. I think ICA auth modules are more interesting when they aren't just checking user private keys but are doing more complicated forms of authentication such as via governance proposals. I would not include the controller module for now. I would wait for others to develop ICA auth modules that could perhaps be reused without accounting for your specific keys